src/epipe.erl

```diff
@@ -13,15 +13,25 @@
       InitialState :: term(),
       PipeFun      :: fun((SomeState :: term()) -> PipeFunRes),
       PipeFunRes   :: {ok, NextState :: term()}
+                    | {stop, State :: term()}
                     | {error, Reason :: term()}
                     | {error, Reason :: term(), NewState :: term()},
       Result       :: {ok, State :: term()}
+                    | {ok, Label, State :: term()}
+                    | {stop, Label, State::term()}
                     | {error, Label, Reason :: term(), NewState :: term()}.
+
 run([], State) ->
     {ok, State};
+run(FuncList, State) when erlang:is_tuple(State) ->
```

Comment:
@raydrawc I would try to keep the epipe more generic. E.g. from my point of view it's not clear why pipe should make assumptions about the State, (e.g. calling tuple_to_list if the State is a tuple, as it seems to be project specific). From our point of view, State is supposed to be opaque for pipe.